modules/gapi/samples/one_vpl_source_to_bgr_conv.cpp

```diff
@@ -0,0 +1,203 @@
+        HRESULT err = CreateDXGIFactory(__uuidof(IDXGIFactory),
+                                    reinterpret_cast<void**>(&out_factory));
+        if (FAILED(err)) {
+            std::cerr << "Cannot create CreateDXGIFactory, error: " << HRESULT_CODE(err) << std::endl;
+            return -1;
+        }
+        adapter_factory = createCOMPtrGuard(out_factory);
+    }
+
+    auto intel_adapter = createCOMPtrGuard<IDXGIAdapter>();
+    UINT adapter_index = 0;
+    const unsigned int refIntelVendorID = 0x8086;
+    IDXGIAdapter* out_adapter = nullptr;
+
+    while (adapter_factory->EnumAdapters(adapter_index, &out_adapter) != DXGI_ERROR_NOT_FOUND) {
+        DXGI_ADAPTER_DESC desc{};
+        out_adapter->GetDesc(&desc);
+        if (desc.VendorId == refIntelVendorID) {
+            intel_adapter = createCOMPtrGuard(out_adapter);
+            break;
+        }
+        ++adapter_index;
+    }
+
+    if (!intel_adapter) {
+        std::cerr << "No Intel GPU adapter on aboard. Exit" << std::endl;
+        return -1;
+    }
+    auto dx11_dev = createCOMPtrGuard<ID3D11Device>();
+    auto dx11_ctx = createCOMPtrGuard<ID3D11DeviceContext>();
+    std::tie(dx11_dev, dx11_ctx) = create_device_with_ctx(intel_adapter.get());
+    void* accel_device_ptr = nullptr;
+    void* accel_ctx_ptr = nullptr;
+    accel_device_ptr = reinterpret_cast<void*>(dx11_dev.get());
+    accel_ctx_ptr = reinterpret_cast<void*>(dx11_ctx.get());
+
+    cv::Ptr<cv::gapi::wip::IStreamSource> cap;
+    std::vector<cv::gapi::wip::onevpl::CfgParam> source_cfgs;
+    source_cfgs.push_back(cfg::create_from_string(
+                                        "mfxImplDescription.AccelerationMode"
+                                        ":"
+                                        "MFX_ACCEL_MODE_VIA_D3D11"));
+    cap = cv::gapi::wip::make_onevpl_src(input, source_cfgs,
+                                         "GPU",
+                                         accel_device_ptr,
+                                         accel_ctx_ptr);
+
+    // Now build the graph
+    cv::GFrame in; // input frame from VPL source
+    auto bgr_gmat = cv::gapi::streaming::BGR(in); // conversion from VPL source frame to BGR UMat
+    auto out = cv::gapi::blur(bgr_gmat, cv::Size(4,4)); // ocl version of blur kernel
+
+    cv::GStreamingCompiled pipeline = cv::GComputation(cv::GIn(in), cv::GOut(out))
+        .compileStreaming(std::move(cv::compile_args(cv::gapi::core::ocl::kernels())));
+    pipeline.setSource(std::move(cap));
+
+    // The execution part
+    size_t frames = 0u;
+    cv::TickMeter tm;
+    cv::Mat outMat;
+
+    pipeline.start();
+    tm.start();
+
+    while (pipeline.pull(cv::gout(outMat))) {
+        cv::imshow("OutVideo", outMat);
+        cv::waitKey(1);
+        ++frames;
+    }
+    tm.stop();
+    std::cout << "Processed " << frames << " frames" << " (" << frames / tm.getTimeSec() << " FPS)" << std::endl;
+#else
+    GAPI_Assert(false && "Built without DX11 support");
```

Comment:
I'd better use 'std::cerr' instead for graceful termination

Reply:
This is definitely shouldn't be in sample

Reply:
Removed from the sample